lib/scholar/metrics.ex

```diff
@@ -369,6 +369,93 @@ defmodule Scholar.Metrics do
     Nx.indexed_add(zeros, indices, updates)
   end
 
+  @doc ~S"""
+  Calculates F1 score given rank-1 tensors which represent
+  the expected (`y_true`) and predicted (`y_pred`) classes.
+
+  ## Options
+
+    * `:num_classes` - required. Number of classes contained in the input tensors
+    * `:average` - optional. This determines the type of averaging performed on the data.
+      * `nil`. Default value. In this case, the scores for each class are returned.
+      * `:macro`. Calculate metrics for each label, and find their unweighted mean.
+        This does not take label imbalance into account.
+      * `:weighted`. Calculate metrics for each label, and find their average weighted by
+        support (the number of true instances for each label).
+      * `:micro`. Calculate metrics globally by counting the total true positives,
```

Comment:
Any references for the naming here?

Reply:
I was referencing to sklearn when I wrote this, [sklearn.metrics.f1_score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html). The example data comes from [here](https://towardsdatascience.com/micro-macro-weighted-averages-of-f1-score-clearly-explained-b603420b292f).

Reply:
Do jax or pytorch have a similar function?

Reply:
I didn't find it in jax or pytorch, but F1 score is a useful measure for imbalanced classification problems.

Reply:
It's in [`torchmetrics`](https://torchmetrics.readthedocs.io/en/stable/classification/f1_score.html).